File: 55.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        gg=0
        n=len(nums)
        for i in range(n):
            if (gg<i):
                return False
            if (i+nums[i]>gg):
                gg=i+nums[i]
            if (gg>=n):
                return True
        return True

File: 55.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int gg=0;
        for (int i=0;i<nums.size();i++)
        {
            if (gg<i)
                return false;
            if (i+nums[i]>gg)
                gg=i+nums[i];
            if (gg>nums.size())
                return true;
        }
        return true;
    }
};
